src/components/Query.tsx

```diff
@@ -26,6 +27,8 @@ class QueryHandler extends Component<QueryHandlerProps, QueryHandlerState> {
   private request = createRequest(this.props.query, this.props.variables);
 
   executeQuery = (opts?: Partial<OperationContext>) => {
+    if (this.props.skip) return;
```

Comment:
When the props change we do call executeQuery again at which point we have to tear down the previous query.

This means that unsubscribe should still be called when skip is true 

The update life cycle method will also have to trigger when skip changes (componentDidUpdate)